src/mrinufft/operators/autodiff.py

```diff
@@ -20,14 +20,20 @@ class _NUFFT_OP(torch.autograd.Function):
     @staticmethod
     def forward(ctx, x, traj, nufft_op):
         """Forward image -> k-space."""
-        ctx.save_for_backward(x, traj)
+        if nufft_op._grad_wrt_data and not nufft_op._grad_wrt_traj:
+            ctx.save_for_backward(x)
+        else :
+            ctx.save_for_backward(x, traj)
```

Comment:
and what if you have `wrt_traj=true` but `wrt_data=False` there is 3 case to check (the 4th, all false is irrelevant)

Reply:
Good catch, i kept thinking that we are missing something surely, but moved focus of my review somewhere else.